web/src/people.ts

```diff
@@ -1501,6 +1504,9 @@ export function make_user(user_id: number, email: string, full_name: string): Us
         // This property allows us to distinguish actual server person
         // objects from fake person objects generated by this function.
         is_missing_server_data: true,
+
+        // Used to differentiate from UserGroup in `user_groups.is_user_group`.
+        members: undefined,
```

Comment:
@andersk this kind of thing feels like maybe an anti-pattern but I wasn't sure. Can you advise?

I also did something similar in the next commit to differentiate mentions from users for the typeahead code, and would love feedback on how I did that.

Reply:
Certainly an anti-pattern. When we have a value that might be either a user or a group, we should use an explicit discriminator to record which: `{type: "user"; user: User} | {type: "group"; group: UserGroup}`.